### PR TITLE
localized_covariance: add xarray wrapper

### LINKED ISSUE #153: covariance & localisation: extract functions
`train_lv_find_localized_ecov` finds the optimal localization radius for the empirical covariance matrix and derives the empirical cov matrix.

https://github.com/MESMER-group/mesmer/blob/3892e764dd80416137a0bf584f316aa48ca1e11b/mesmer/calibrate_mesmer/train_lv.py#L273

The statistical methods should be extracted.

### PR DESCRIPTION
**Pull request by mathause**
- [x] Closes #153
 - [x] Tests added
 - [x] Passes `isort . && black . && flake8`
 - [ ] Fully documented, including `CHANGELOG.rst`

xarray wrapper for `find_localized_empirical_covariance` and `adjust_covariance_ar1`. Ugly as hell but there is no good way to have the same dim more than once for a DataArray.

cc @znicholls

### CODE REVIEW

File: mesmer/core/localized_covariance.py [hunk 1]
```diff
@@ -63,6 +75,79 @@ def _adjust_ecov_ar1_np(covariance, ar_coefs):
     return reduction_factor * reduction_factor.T * covariance
 
 
+def find_localized_empirical_covariance(data, weights, localizer, dim, k_folds):
+    """determine localized empirical covariance by cross validation
+
+    Parameters
+    ----------
+    data : 2D array
+        Data array with shape n_samples x n_gridpoints.
```
Comment (mathause) [inline mesmer/core/localized_covariance.py:84]:
```suggestion
    data : xr.DataArray
        2D DataArray with with n_samples x n_gridpoints.
```

File: mesmer/core/localized_covariance.py [hunk 2]
```diff
@@ -63,6 +75,79 @@ def _adjust_ecov_ar1_np(covariance, ar_coefs):
...
+    weights : 1D array
```
Comment (mathause) [inline mesmer/core/localized_covariance.py:85]:
```suggestion
    weights : xr.DataArray
```

File: mesmer/core/localized_covariance.py [hunk 3]
```diff
@@ -63,6 +75,79 @@ def _adjust_ecov_ar1_np(covariance, ar_coefs):
...
+        Weights for the individual samples.
+    localizer : dict of array-like
```
Comment (mathause) [inline mesmer/core/localized_covariance.py:87]:
```suggestion
    localizer : dict of DataArray```

File: mesmer/core/localized_covariance.py [hunk 4]
```diff
@@ -63,6 +75,79 @@ def _adjust_ecov_ar1_np(covariance, ar_coefs):
...
+        Dictonary containing the localization radii as keys and the localization matrix
+        as values. The localization must be 2D and of shape nr_gridpoints x nr_gridpoints.
```
Comment (mathause) [inline mesmer/core/localized_covariance.py:89]:
```suggestion
        as values. The localization must be 2D and of shape n_gridpoints x n_gridpoints.
```

**Comment by mathause** [comment]
@znicholls do you have time to take a look at this?

**Review by znicholls — COMMENTED** [review]
Sorry I looked at this last night but forgot to hit 'Submit'

**Comment by mathause** [comment]
Thanks for your review!

**Pull request merged**